azurerm/internal/services/logic/validate/logic.go

```diff
@@ -28,3 +29,25 @@ func IntegrationAccountID(i interface{}, k string) (warnings []string, errors []
 
 	return warnings, errors
 }
+
+// https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules
+// 1-80	Alphanumerics, hyphens, periods, and underscores.
+func IntegrationServiceEnvironmentName() schema.SchemaValidateFunc {
+	return validation.StringMatch(
+		regexp.MustCompile(`^[\w-.]{1,80}$`), `Integration Service Environment name can contain only letters, numbers, '_','-' or '.' up to 80 characters in length.`,
+	)
+}
+
+func ValidateSubnetID(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
+		return warnings, errors
+	}
+
+	if _, err := networkParse.SubnetID(v); err != nil {
+		errors = append(errors, fmt.Errorf("Can not parse %q as a subnet id: %v", k, err))
```

Comment:
Per the go language standards the err msg should not contain capitalization...

```suggestion
		errors = append(errors, fmt.Errorf("can not parse %q as a subnet id: %v", k, err))
```

Reply:
Thanks, will change it.